core/src/main/java/inetsoft/uql/asset/internal/MirrorAssemblyImpl.java

```diff
@@ -411,15 +411,15 @@ public synchronized void clearCache() {
 
    public String handleWSOrgMismatch(String identifier){
       String curOrgId = OrganizationManager.getInstance().getCurrentOrgID();
-      int orgIdx = identifier.lastIndexOf("^");
+      int numCarets = identifier.length() - identifier.replace("^", "").length();
```

Comment:
Doing a string replacement is unnecessarily complex for this use. Use `StringUtils.countMatches(identifier, '^')` instead.